src/components/Login.js

```diff
@@ -0,0 +1,19 @@
+import { useEffect } from 'react';
+import { useHistory } from 'react-router-dom';
+
+function Login() {
+  const history = useHistory();
+  useEffect(() => {
+    const searchParams = new URL(window.location).searchParams;
+    const accessToken = searchParams.get('access_token');
+    if (accessToken) {
+      localStorage.setItem('GITHUB_ACCESS_TOKEN', accessToken);
+    }
+    console.log(accessToken);
+    history.push('/');
+  });
```

Comment:
you seem to be missing the dependency array here :)

Reply:
I am aware of that, thank you! :) I finally managed to sort out my personal stuff and I will be finishing this draft in following days :)